api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java

```diff
@@ -220,6 +228,13 @@ public List<Appointment> getAllAppointmentsReminder(String hours) {
 
     }
 
+    @Transactional
```

Comment:
I personally think this should be `@Transactional(readOnly = true)`, but I'm fine just doing what you are doing to be consistent with the rest of the file I guess, and then we can create a new ticket to clean all of this up.

I do think that, since you have this new service impl method, that you should change the existing `getAllAppointments(Date)` method to delegate to this one, just like you did in the DAO.